ui/src/app/components/buttons.tsx

```diff
@@ -1,93 +1,77 @@
-  }),
-  secondary: ({ hover, disabled }) => ({
-    border: '2px solid',
-    borderColor: disabled ? '#c3c3c3' : '#262262',
-    backgroundColor: disabled ? '#f1f2f2' : (hover ? '#262262' : '#ffffff'),
-    borderRadius: '0.2rem',
-    color: disabled ? '#c3c3c3' : (hover ? '#ffffff' : '#262262'),
-  }),
-  darklingPrimary: ({ hover, disabled }) => ({
-    backgroundColor: disabled ? '#c3c3c3' : (hover ? 'rgba(255,255,255,0.3)' : '#262262'),
-    color: '#ffffff',
-    border: '0px',
-    borderRadius: '0.2rem'
-  }),
-  darklingSecondary: ({ hover, disabled }) => ({
-    backgroundColor: disabled ? '#c3c3c3' : (hover ? '#50ACE1' : '#0079b8'),
-    color: '#ffffff',
-    border: '0px',
-    borderRadius: '0.2rem'
-  })
-};
-
-/*
- * Uses the function-as-child technique. The children function is called with:
- *   hover: the current hover state
- *   trackHover: transforms an element, adds hooks to track hover state
- */
-class HoverContainer extends React.Component<
-  { children: ({ hover: boolean, trackHover: Function }) => React.ReactNode },
-  { hover: boolean }
-> {
-  constructor(props) {
-    super(props);
-    this.state = { hover: false };
   }
+};
 
-  trackHover = el => {
-    return React.cloneElement(el, {
-      onMouseEnter: (...args) => {
-        if (el.props.onMouseEnter) {
-          el.props.onMouseEnter(...args);
-        }
-        this.setState({ hover: true });
-      },
-      onMouseLeave: (...args) => {
-        if (el.props.onMouseLeave) {
-          el.props.onMouseLeave(...args);
-        }
-        this.setState({ hover: false });
-      }
-    });
+const buttonVariants = {
+  primary: {
+    style: {
+      ...styles.base,
+      borderRadius: '0.3rem',
+      backgroundColor: '#262262', color: '#fff',
+    },
+    disabledStyle: {backgroundColor: '#c3c3c3'},
+    hover: {backgroundColor: '#4356A7'}
+  },
+  secondary: {
+    style: {
+      ...styles.base,
+      border: '2px solid', borderRadius: '0.2rem', borderColor: '#262262',
+      backgroundColor: '#ffffff', color: '#262262',
+    },
+    disabledStyle: {
+      borderColor: '#c3c3c3',
+      backgroundColor: '#f1f2f2', color: '#c3c3c3'
+    },
+    hover: {backgroundColor: '#262262', color: '#ffffff'}
+  },
+  darklingPrimary: {
+    style: {
+      ...styles.base,
+      borderRadius: '0.2rem',
+      backgroundColor: '#262262', color: '#ffffff'
+    },
+    disabledStyle: {backgroundColor: '#c3c3c3'},
+    hover: {backgroundColor: 'rgba(255,255,255,0.3)'}
+  },
+  darklingSecondary: {
+    style: {
+      ...styles.base,
+      borderRadius: '0.2rem',
+      backgroundColor: '#0079b8', color: '#ffffff'
+    },
+    disabledStyle: {backgroundColor: '#c3c3c3'},
+    hover: {backgroundColor: '#50ACE1'}
   }
+};
 
-  render() {
-    const { children } = this.props;
-    const { hover } = this.state;
-    return children({ hover, trackHover: this.trackHover });
-  }
-}
+const computeStyle = ({style, hover, disabledStyle}, {disabled}) => {
```

Comment:
Lou's comment in [slack](https://pmi-engteam.slack.com/archives/C66DX9FD0/p1547221639134600) indicates that we might need a key-pressed state, but that's likely separate from this PR. Just food for thought.